src/osd/osd_op_util.cc

```diff
@@ -108,12 +120,18 @@ int OpInfo::set_from_op(
       if (ceph_osd_op_mode_modify(iter->op.op))
 	set_write();
     }
-    if (ceph_osd_op_mode_read(iter->op.op))
+    if (ceph_osd_op_mode_read(iter->op.op)) {
       set_read();
+      if (iter->op.op != CEPH_OSD_OP_STAT) {
+        set_read_data();
+      }
+    }
 
     // set READ flag if there are src_oids
-    if (iter->soid.oid.name.length())
+    if (iter->soid.oid.name.length()) {
```

Comment:
The above `src_oids` comment is hopelessly outdated -- in fact this entire `if` statement is probably dead code.

Reply:
I have submitted https://github.com/ceph/ceph/pull/50499 to clean that up.